test/unit/utils.test.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const { expect } = require('chai');
+const utils = require('../../src/utils');
+
+describe('utils.js', () => {
+  describe('exports', () => {
+    it('should have toLegacy symbol', () => {
+      expect(utils).to.have.property('toLegacy').that.is.a('symbol');
+    });
+
+    it('should have maybeCallback helper', () => {
+      expect(utils).to.have.property('maybeCallback').that.is.a('function');
+    });
+  });
+
+  describe('maybeCallback', () => {
+    const maybeCallback = utils.maybeCallback;
+    it('should accept to two arguments', () => {
+      expect(maybeCallback).to.have.lengthOf(2);
+    });
+
+    it('should return promise provided if no other arguments are present', async () => {
+      const promise = Promise.resolve(2);
```

Comment:
At the very least, let's group these tests by whether or not they take a callback.  Optional, but preferable, group them by context

```typescript
context('when no callback is provided', function() {
  context('when the promise resolves', () => { 
    it('returns the resolved promise unmodified', () => { ... })
  })
  context('when the promise rejects', () => { 
    it('returns the rejected promise', () => { ... })
  })
})

context('when callback is provided', function() {
    context('when the promise resolves', () => { 
        it('calls the callback with the result from the promise', () => { ... })
    })

    context('when the promise rejects', () => { 
        it('calls the callback with the error and no value', () => { ... })
    })
})
```